plugins/msr/fu-plugin-msr.c

```diff
@@ -34,6 +37,21 @@ fu_plugin_init (FuPlugin *plugin)
 	fu_plugin_add_udev_subsystem (plugin, "msr");
 }
 
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+	guint eax = 0;
+	guint ebx = 0;
+	guint ecx = 0;
+	guint edx = 0;
+
+	/* sdbg is supported: https://en.wikipedia.org/wiki/CPUID */
+	__get_cpuid (0x01, &eax, &ebx, &ecx, &edx);
+	priv->is_supported = ((ecx >> 11) & 0x1) > 0;
+	return TRUE;
```

Comment:
why not just abort plugin startup here?

Reply:
good question, i did ponder outputting the status as "not supported" but changed by mind. Do you think not-suported makes sense, or just not showing anything about DCI?

Reply:
I think showing `not supported` just begs the questions:
* what do I do to support it?  
* Is that a good thing it's not supported?
* Is something wrong with my system?

Since the system doesn't support it I think just return FALSE in that circumstance and leave it out of the equation.  Here's the perfect example:
* If you're running on AMD and AMD doesn't support something like DCI, would you show not supported?  I think the answer is no.

Reply:
The other thing I wondered was when we add more things to the MSR plugins, only some of the HSI checks need to be shipped. I guess we could rename `is_supported` to `ia32_debug_supported` and then skip the tests in the two `ia32_debug`-specific tests.

Reply:
that makes sense to me